### PR TITLE
`Development`: Use AssertJ for assertions in server tests

### PR DESCRIPTION
**Pull request by laadvo**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/)
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Many tests use the default JUnit assertions instead of AssertJ, a powerfull library that offers many different assertion. It is considered to be best practice to use AssertJ, since it provides descriptive error messages.

### Description

In this PR `assertTrue`, `assertFalse`, `assertEquals`, `assertNull` and `assertNotNull` are replaced with their AssertJ equivalents. The PR does not cover `assertThrows`,  this will be done in a follow-up. I also fixed a few instances where the expected and actual value were switched.

### Review Progress

Look through the code and make sure the most specific assertions have been used. Also check if there are some assertions that might have been missed.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage
unchanged

### CODE REVIEW

**Review by b-fein — DISMISSED** [review]

**Review by terlan98 — COMMENTED** [review]
Code LGTM except for one line where `assertNotNull` is not replaced 🥲

File: src/test/java/de/tum/in/www1/artemis/service/exam/ExamServiceTest.java [hunk 1]
```diff
@@ -122,7 +122,7 @@ void validateForStudentExamGeneration_differentCalculationTypesInExerciseGroup_s
         BadRequestAlertException thrown = assertThrows(BadRequestAlertException.class, () -> examService.validateForStudentExamGeneration(exam),
                 "Expected to throw bad request alert exception, but it didn't");
 
-        assertTrue(thrown.getMessage().contains("All exercises in an exercise group must have the same meaning for the exam score"));
+        assertThat(thrown.getMessage()).contains("All exercises in an exercise group must have the same meaning for the exam score");
```
[the remark this answers is not in the document]
Comment (laadvo) [reply src/test/java/de/tum/in/www1/artemis/service/exam/ExamServiceTest.java]: Oh, thanks for pointing that out! I found other similar occurences that could use `.hasMessage...` and will fix them in the follow-up :+1:

File: src/test/java/de/tum/in/www1/artemis/connectors/Lti13ServiceTest.java [hunk 2]
```diff
@@ -461,20 +462,20 @@ void onNewResult() throws ParseException {
 
         List<String> authHeaders = httpEntity.getHeaders().get("Authorization");
         assertNotNull(authHeaders, "Score publish request must contain an Authorization header");
```
[the remark this answers is not in the document]
Comment (laadvo) [reply src/test/java/de/tum/in/www1/artemis/connectors/Lti13ServiceTest.java]: Good catch, fixed it now :)

**Review by tobias-lippert — APPROVED** [review]
Code LGTM, only found instances of assertThrows, assertDoesNotThrow and fail on the branch. 
Maybe you could also test this e.g. with an additional test in the ArchitectureTest class
``` 
@Test
    void testNoJunitJupiterAssertions() {
        ArchRule noJunitJupiterAssertions = noClasses().should().dependOnClassesThat().haveNameMatching("org.junit.jupiter.api.Assertions");
        noJunitJupiterAssertions.check(testClasses);
    }
```

**Comment by laadvo** [comment]
> Code LGTM, only found instances of assertThrows, assertDoesNotThrow and fail on the branch. Maybe you could also test this e.g. with an additional test in the ArchitectureTest class
> 
> ```
> @Test
>     void testNoJunitJupiterAssertions() {
>         ArchRule noJunitJupiterAssertions = noClasses().should().dependOnClassesThat().haveNameMatching("org.junit.jupiter.api.Assertions");
>         noJunitJupiterAssertions.check(testClasses);
>     }
> ```

That's a really good idea :)   
I will add it in the next PR, where last JUnit assertions still remaining will be replaced.

**Review by terlan98 — APPROVED** [review]
Code LGTM, thanks for the fix 👍

**Review by maximiliansoelch — APPROVED** [review]

**Pull request merged**